### PR TITLE
Enable `ValidatedSanitizedInput` rule in plugin review check

### LINKED ISSUE #582: Check: Input data must be Sanitized and Validated
We need this check as it should be at the top of the list of security checks. Essentially when the plugin includes POST/GET/REQUEST/FILE calls, must be sanitized, validated and escaped.

Our main goal is to prevent a user from accidentally sending trash data through the system, as well as protecting them from potential security issues.

We have to look for sniffs from WPCS, maybe `ValidatedSanitizedInputSniff` could be one of the best for this purpose.

### PR DESCRIPTION
**Pull request by davidperezgar**
Solves #582

### CODE REVIEW

**Comment by davidperezgar** [comment]
Finished and ready for review

File: phpcs-rulesets/plugin-review.xml [hunk 1]
```diff
@@ -46,6 +46,12 @@
 		<type>warning</type>
 	</rule>
 
+	<!-- Sanitized Input rules -->
+	<rule ref="WordPress.Security.ValidatedSanitizedInput">
+		<type>error</type>
+		<severity>7</severity>
+	</rule>
```
Comment (swissspidy) [inline phpcs-rulesets/plugin-review.xml:53]: When we add `ValidatedSanitizedInput` to this file here, my understanding is that the `Plugin_Review_PHPCS_Check` check will run it.

So there's no need for `Sanitized_Input_Check`, as `Plugin_Review_PHPCS_Check` already scans for this. Or am I missing something?

Comment (ernilambar) [reply phpcs-rulesets/plugin-review.xml:53]: Agree. Just adding these rules in the ruleset should trigger the errors.

Comment (davidperezgar) [reply phpcs-rulesets/plugin-review.xml:53]: Yes, you're right. I thought that I've to create a file check as well. I've changed and updated.

File: tests/phpunit/tests/Checker/Checks/Sanitized_Input_Check_Tests.php [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+ * Tests for the Sanitized_Input_Check class.
+ *
+ * @package plugin-check
+ */
+
+use WordPress\Plugin_Check\Checker\Check_Context;
+use WordPress\Plugin_Check\Checker\Check_Result;
+use WordPress\Plugin_Check\Checker\Checks\Security\Sanitized_Input_Check;
+
+class Sanitized_Input_Check_Tests extends WP_UnitTestCase {
```
Comment (swissspidy) [inline tests/phpunit/tests/Checker/Checks/Sanitized_Input_Check_Tests.php]: We can now remove this test in favor of `Plugin_Review_PHPCS_Check_Tests`

The test files you added can be merged into `test-plugin-review-phpcs-errors/load.php` / `test-plugin-review-phpcs-without-errors/load.php`

Comment (davidperezgar) [reply tests/phpunit/tests/Checker/Checks/Sanitized_Input_Check_Tests.php]: ok

File: tests/phpunit/tests/Checker/Checks/Plugin_Review_PHPCS_Check_Tests.php [hunk 3]
```diff
@@ -22,7 +22,7 @@ public function test_run_with_errors() {
 
 		$this->assertNotEmpty( $errors );
 		$this->assertArrayHasKey( 'load.php', $errors );
-		$this->assertEquals( 2, $check_result->get_error_count() );
+		$this->assertEquals( 5, $check_result->get_error_count() );
```
Comment (swissspidy) [inline tests/phpunit/tests/Checker/Checks/Plugin_Review_PHPCS_Check_Tests.php:25]: You added 1 line to the test file, why did the error count increase from 2 to 5? 🤔

Comment (ernilambar) [reply tests/phpunit/tests/Checker/Checks/Plugin_Review_PHPCS_Check_Tests.php:25]: It triggers 3 errors under that same sniff.

**Comment by davidperezgar** [comment]
Ready for approval!

**Review by swissspidy — APPROVED** [review]

**Review by ernilambar — APPROVED** [review]

**Pull request merged**